internal/rbd/replicationcontrollerserver.go

```diff
@@ -437,6 +517,32 @@ func (rs *ReplicationServer) PromoteVolume(ctx context.Context,
 		}
 	}
 
+	var mode librbd.ImageMirrorMode
+	mode, err = getMirroringMode(ctx, req.GetParameters())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get mirroring mode %s", err.Error())
+	}
+
+	log.DebugLog(ctx, "Attempting to tickle dummy image for restarting RBD schedules")
+	err = tickleMirroringOnDummyImage(rbdVol, mode)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to enable mirroring on dummy image %s", err.Error())
+	}
+
+	interval, startTime := getSchedulingDetails(req.GetParameters())
+	if interval != admin.NoInterval {
+		err = rbdVol.addSnapshotScheduling(interval, startTime)
```

Comment:
@ShyamsundarR @Madhu-1 we have a bug in this line. `rbdVol` was changed to `rbdVol dummy` object in line 527 changed.  In this line, we still point to the same dummy rbdVol.  Adding schedule here is adding it to the wrong image.  I didn't catch this in my testing because my real image also has a schedule of 1m.  If you change the real image schedule to something else, you will see the dummy image schedule added with "something else".

@Madhu-1 It is an easy fix.  Can you make the change?

Reply:
got it. tested the fix making a PR.

Reply:
Thanks @Madhu-1 for providing the fix so quickly.  I have tested patch #2669 and it works as intended.